inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/utils/BufferQueue.java

```diff
@@ -61,6 +61,19 @@ public A pollRecord() {
         return record;
     }
 
+    /**
+     * Take record
+     */
+    public A takeRecord() {
+        try {
+            A record = queue.take();
+            this.offerCount.incrementAndGet();
```

Comment:
It is take operation, need to invoke
this.pollCount.getAndIncrement();

Reply:
done